### ISSUE TITLE
Project : Dataset and clearity

### ISSUE THREAD

**Issue by anshumitts** [vineetm/ell-881-2018-deep-learning#13]
Hi, There is a bit confusion,
1. On which dataset should we be training our model?
2. on Task 10.2,10.3 should we re-train the embeddings or else which model should we use to perform the task
3. I am not sure what is this "SICK" in 10.4

**Comment by vineetm** [comment]
@anshumitts Thanks for your question.

Let me clarify the confusion. The evaluation tasks 10.1-10.4 are **independent** of training your RNN models.

Let us for sake of clarity make a distinction between tasks. 
**Training Tasks**: There are three independent training tasks: NLI, Constituency Parsing & NMT on En-De. These are related only by sharing RNN parameters. While training you would only use the corresponding training/dev data. For example for NLI Training Task, you will use the ALLNLI dataset (with 1M pairs). For dependency parsing, you will use Parsing dataset. These are summarized in Table1
**Evaluation Tasks**: Once your models are trained, you will move to the evaluation tasks, where you will use the learnt RNN Model to perform these tasks (10.1-10.4). For details on what are the evaluation tasks such as SICK look at the [paper github repo](https://github.com/Maluuba/gensen) which points to SentEval used to evaluate the models

I would not worry much about 10.1-10.4 to begin with. First, you should get a handle on **Training Tasks**.

Please feel free to ask any followup questions.

**Comment by anshumitts** [comment]
Hi, @vineetm  Can you please provide dataset for parse tree as no dataset has been provided which was fed to PUCK GPU to get noise dataset and they have updated penntree bank dataset. So I am not sure I am getting the correct parse tree.

**Comment by anshumitts** [comment]
@vineetm  can we use a different constituency parser? as I cannot setup puck on HPC.

**Comment by vineetm** [comment]
@Maxaravind @VinayKyatham can you please look into this and see why @anshumitts cannot setup this on HPC. Here is the link https://github.com/dlwh/puck

**Comment by Maxaravind** [comment]
@anshumitts Can you tell me what is the exact problem that you are facing?

**Comment by Maxaravind** [comment]
Is this the full trace?

**Comment by Maxaravind** [comment]
@anshumitts While running jobs on hpc please 'cd' into the working directory in your qsub script, otherwise relative names won't work. Looking at the trace, it seems to be a problem with your Current Working Directory not being set properly

**Comment by abhudev** [comment]
@anshumitts @Maxaravind Any updates on how to make this work? Even I could not make it work

**Comment by vineetm** [comment]
@Maxaravind  @VinayKyatham can you please look into this? @anshumitts @abhudev I would suggest you try first by running the parser on some sample sentences say 100 (do some kind of hello world equivalent).

**Comment by abhudev** [comment]
@vineetm I think the issue is that of the correct libraries etc being in place e.g. the error logs mention OpenCL, which is not there in the list of HPC modules. Also, if we were to do this on our machines, I doubt it would run (or run fast, as many of us don't have GPU's).

**Comment by vineetm** [comment]
@abhudev Can you try running this on your local system? Slow is better than nothing :) Let me know if you can run it locally. Meanwhile @Maxaravind  @VinayKyatham will help us debug what is missing on HPC.

**Comment by abhudev** [comment]
@vineetm Yes, I will see if I can get it to work on my PC.

**Comment by Maxaravind** [comment]
@abhudev @anshumitts  Please make sure that you are able to run your code on your local machine before you run it on hpc(and make a list of dependencies that you might require). Hpc provides a very restricted environment and it would be good to make sure that your code is not working just because of the hpc environment. Please do let us know whether you are able to get it working on your PC.

**Comment by vineetm** [comment]
@Maxaravind  @VinayKyatham It would be better if you can also try to run the parser on (1) HPC (2) your local system so you can help students get this up and running. Please let me know when you are able to do so. I would suggest you try with a small set of sentences to check if it works.. Thanks!

**Comment by vineetm** [comment]
@abhudev @anshumitts While @Maxaravind  @VinayKyatham help us debug PUCK. You can look at some alternatives for constituency parsing. Given the short time left for deadline, I would not want you to put in more time in figuring out how the software works. Have a look at Issue #20

**Comment by abhudev** [comment]
Yes Sir, I will try the other two parsers if they are working OK
